### PR TITLE
add attestation endpoint and token audience required for authenticating with Azure Attestation tenants

### PR DESCRIPTION
**Pull request by shleiAmy**

### CODE REVIEW

**Comment by shleiAmy** [comment]
@isra-fel could you please help review this pr? Thank you

File: src/Authentication.Abstractions/Interfaces/IAzureEnvironment.cs [hunk 1]
```diff
@@ -131,5 +131,15 @@ public interface IAzureEnvironment : IExtensibleModel
         /// The set of version profile s(service capabilities) supported
         /// </summary>
         IList<string> VersionProfiles { get; }
+
+        /// <summary>
+        /// The domain name suffix for Azure Attestation tenants
+        /// </summary>
+        string AzureAttestationDnsSuffix { get; set; }
```
Comment (isra-fel) [inline src/Authentication.Abstractions/Interfaces/IAzureEnvironment.cs]: Adding things to the interface should be fine... have you test it with the code in azure-powershell?

Comment (shleiAmy) [reply src/Authentication.Abstractions/Interfaces/IAzureEnvironment.cs]: I have hard code AttestationDnsSuffix to attest.azure.net in azure-powershell repo to test the data plane operations, it works

Comment (isra-fel) [reply src/Authentication.Abstractions/Interfaces/IAzureEnvironment.cs]: Let me see if I can build a package of common code for you to test

**Pull request merged**